src/api/controller/AuthenticationController.ts

```diff
@@ -0,0 +1,51 @@
+import { Request, Response } from "express";
+import { IUserDatabase } from "../../database/IUserDatabase";
+import { Validator } from "../../utils/Validator";
+import { UserSchema } from "../model/user/UserSchema";
+
+export class AuthenticationController {
+    private database: IUserDatabase;
+
+    constructor(database: IUserDatabase) {
+        this.database = database;
+    }
+
+    login = async (req: Request, res: Response) => {
+        let existingUser : any = await this.database.GetUser(new Map([["uid", req.uid!]]));
+
+        if (existingUser === null)
+            res.status(400).json({ success: false, data: `user doesn't exists` });
+        else 
+            res.redirect(302, `/api/user/${existingUser._id}`);
+    }
+
+    register = async (req: Request, res: Response) => {
+        const newUser =
+            new UserSchema(
+                req.body.firstName,
+                req.body.lastName,
+                req.uid === undefined ? "" : req.uid
+            );
+
+        const validator = new Validator<UserSchema>();
+
+        let logs = await validator.validate(newUser);
+
+        if (logs.length > 0)
+            res.status(400).json({ success: false, data: logs });
+        else {
+            let existingUser = await this.database.GetUser(new Map([["uid", newUser.uid]]));
+
+            if (existingUser !== null)
+                res.status(400).json({ success: false, data: `uid ${newUser.uid} already exists` });
+            else {
+                let createdUser = await this.database.CreateUser(newUser);
+
+                if (createdUser === null)
+                    res.status(400).json({ success: false, data: newUser });
```

Comment:
Why do we return newUser on failure?

Reply:
Well, my thought process was that maybe we need to return the info about the user that was attempted to be created. I can return nothing if you want me to